### ISSUE TITLE
Add Psalter support to fix UnusedVariable errors

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1926]
Breaking out from #1442

@souravchandra wrote

For `UnusedVariable`

maybe we should just remove the variable if possible
```php
$response = $api->request(...);
```
can be converted to 
```php
$api->request(...);
```
One tricky situation might be where we define something like
```php
// $headers is used but $data is not
$headers = $data = [];
```
and it should be
```php
$headers = [];
```

**Comment by muglug** [comment]
We also want to remove

```
$unused_variable = null;
$unused_variable = 5;
$unused_variable = true;
$unused_variable = [];
$unused_variable = $another_var;
```

I imagine it's sufficient to use `token_get_all` from the last variable character to the next `;`, and check the statements in between.

**Comment by muglug** [comment]
Fixed by #1967
